### DISCUSSION TITLE
Multiple spans created in SQL format

### DISCUSSION THREAD

**Discussion by nryberg** [facelessuser/pymdown-extensions#2313] [Q&A] [answered]
### Description

When I apply Superfences and inlinehilite to SQL, the text gets colorized, which is great.

However, it inserts multiple empty <span></span> blocks into the output.  I'd like to keep it tight together without the multiple breaks.

Have:

![have](https://github.com/facelessuser/pymdown-extensions/assets/46165/6280b840-f78f-4b16-9175-7c0d8532301a)

Want: 

![want](https://github.com/facelessuser/pymdown-extensions/assets/46165/d8ee2801-a047-429c-b0aa-b1a90f25f048)

For this statement:
`
```sql
SELECT bob, fred, george FROM sample
```
`

### Minimal Reproduction

1. Create Markdown text with fenced SQL using triple back ticks
2. Apply extensions 
`markdown.markdown(markdown_text, extensions=['tables', 'pymdownx.superfences', 'pymdownx.inlinehilite'])
`
3. Results very spread out visual.

![highlight](https://github.com/facelessuser/pymdown-extensions/assets/46165/5557289c-be0b-4dcd-b0c1-f1a44559472c)

### Version(s) & System Info

- Operating System: Mac O/S
- Python Version: 3.10.11
- Package Version: 10.7

**Comment by facelessuser** [comment, marked as answer]
> Results very spread out visual.

Neither SuperFences nor InlineHilite inherently exhibit the issue you describe. If this issue exists, it exists because you have provided bad CSS that does this.

The highlighting is controlled by [Pygments](https://pygments.org/), not us. They wanted to properly scope the white space as not being part of the other syntax. Scoping the white space with a span does not inject new lines, that's not how HTML works. But CSS can change how blocks are styled. And since Pymdown Extensions does not provide any CSS in our output, the bad CSS has to originate from either you or some other 3rd party CSS you are including.

Here I enter inline and block format into my documents via:

````markdown
`#!sql SELECT bob, fred, george FROM sample`

```sql
SELECT bob, fred, george FROM sample
```
````

And this is what I get in my documents:

<img width="799" alt="Screenshot 2024-02-08 at 6 47 01 AM" src="https://github.com/facelessuser/pymdown-extensions/assets/1055125/d88161d1-69ed-4e26-9f6a-60bc0144f896">

You can try it out here on CodePen, the content is not spread out: https://codepen.io/facelessuser/pen/WNmgGdK.

**Reply by facelessuser** [reply]
As there is no actual issue, converting this to a discussion.

**Reply by nryberg** [reply]
Makes sense - thanks for the feedback and correction - I'll go back to the CSS and see where it went off the rails.
